kvui.py

```diff
@@ -770,6 +774,42 @@ def fix_heights(self):
             element.height = max_height
 
 
+class ApAsyncImage(AsyncImage):
+    def is_uri(self, filename):
+        if filename[:3] == "ap:":
+            return True
+        else:
+            return super().is_uri(filename)
+
+
+class ImageLoaderPkgutil(ImageLoaderBase):
+    def load(self, filename):
+        # take off the "ap:" prefix
+        module, path = filename[3:].split("/", 1)
+        data = pkgutil.get_data(module, path)
+        return self._bytes_to_data(data)
+
+    def _bytes_to_data(self, data):
+        from PIL import Image as PImage
+        p_im = PImage.open(io.BytesIO(data)).convert("RGBA")
+        im_d = ImageData(p_im.size[0], p_im.size[1], p_im.mode.lower(), p_im.tobytes())
+        return [im_d]
+
+
+# grab the default loader method so we can override it but use it as a fallback
+DefaultLoad = ImageLoader.load
+
+
+def load_override(filename, default_load=DefaultLoad, **kwargs):
+    if filename[:3] == "ap:":
+        return ImageLoaderPkgutil(filename)
+    else:
+        return default_load(filename, **kwargs)
```

Comment:
If we're overriding `load`, why aren't we returning the same type that `load` returns?
It looks like the default `load` returns an instance of `Image`, while the new `load` would return an instance of `ImageLoaderBase` (which itself has a `load` function that returns a list of `ImageData`).
Can you explain how this works?

Reply:
if ImageLoader.load() needs to load asynchronously it returns an ImageLoaderBase that can handle the request
https://github.com/kivy/kivy/blob/master/kivy/core/image/__init__.py#L454
this is mimicking that workflow but just short-circuiting it as early as possible because the current loader picking code in kivy cannot handle the information we need in the filename